### PR TITLE
#359: Added payable interaction test

### PR DESCRIPTION
**Pull request by rusiboyadjiev**

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractCallSuite.java [hunk 1]
```diff
@@ -175,6 +176,46 @@ private HapiApiSpec htsTransfer() {
...
+				.given(
+						/* crypto create */
+						cryptoCreate(RECIPIENT)
+								.balance(RECIPIENT_BALANCE),
+						/* contract creation */
+						fileCreate("bytecode")
+								.path(ContractResources.PAYABLE_INTERACTION),
+						contractCreate("payableInteraction")
+								.bytecode("bytecode")
+								.via("payableInteractionTx")
```
Comment (failfmi) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractCallSuite.java:197]: This is not needed

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractCallSuite.java [hunk 2]
```diff
@@ -175,6 +176,46 @@ private HapiApiSpec htsTransfer() {
...
+						}),
+						sourcing(() ->
+								contractCall(
+										"payableInteraction",
+										ContractResources.PAYABLE_INTERACTION_ABI,
+										asSolidityAddressHex(recipient.get())
+								)
+										.sending(TRANSFER_AMOUNT)
+										.gas(GAS_LIMIT)
+										.via("payableInteractionContract")
```
Comment (failfmi) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractCallSuite.java:212]: This is not needed

**Review by failfmi — APPROVED** [review]

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractCallSuite.java (hunk 2 above)
Comment (rusiboyadjiev) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractCallSuite.java]: removed

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractCallSuite.java (hunk 1 above)
Comment (rusiboyadjiev) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ContractCallSuite.java]: removed

**Review by Daniel-K-Ivanov — APPROVED** [review]

**Pull request merged**